apps/rule-manager/app/service/RuleManager.scala

```diff
@@ -176,6 +176,18 @@ object RuleManager extends Loggable {
             )
           )
         )
+      _ <- draftRule match {
+        case _ if draftRule.isArchived =>
+          Left(
+            Seq(
+              FormError(
+                "Rule is archived",
+                "Only unarchived rules can be published"
+              )
+            )
+          )
+        case _ => Right(())
+      }
```

Comment:
The server side validation for publish is a little more complex than for the other state transitions, where we simply check for an if condition in a for comprehension.